src/dlgTriggerEditor.cpp

```diff
@@ -780,386 +952,761 @@ void dlgTriggerEditor::slot_item_selected_search_list(QTreeWidgetItem* pItem, in
+                        setAllSearchData(parent, cmKeysView, name, key->getID(), SearchResultIsScript, startPos, index, instance++);
+                        treeWidget_searchResults->addTopLevelItem(parent);
+                    } else {
+                        sl << QString() << QString() << tr("Lua code {L: %1 C: %2}").arg(index+1).arg(startPos+1) << whatText;
+                        pItem = new QTreeWidgetItem(parent, sl);
+                        pItem->setFirstColumnSpanned(false);
+                        setAllSearchData(pItem, cmKeysView, name, key->getID(), SearchResultIsScript, startPos, index, instance++);
+                        parent->addChild(pItem);
+                        parent->setExpanded(true);
+                    }
+                    ++startPos;
+                }
+            }
+
             recursiveSearchKeys(key, s);
         }
     }
 
-    if (true) {
+    {
         if (mCurrentView != cmVarsView) {
+            // repopulateVars can take some time should there be a large number
+            // of variables or big tables... 8-(
             repopulateVars();
         }
+
         LuaInterface* lI = mpHost->getLuaInterface();
         VarUnit* vu = lI->getVarUnit();
         TVar* base = vu->getBase();
-        QListIterator<TVar*> it(base->getChildren(false));
-        while (it.hasNext()) {
-            TVar* var = it.next();
+        QListIterator<TVar*> itBaseVarChildren(base->getChildren(false));
+        while (itBaseVarChildren.hasNext()) {
+            TVar* var = itBaseVarChildren.next();
+            // We do not search for hidden variables - probably because we would
+            // have to unhide all of them to show the hidden ones found by
+            // searching
             if (!showHiddenVars && vu->isHidden(var)) {
                 continue;
             }
+
             //recurse down this variable
             QList<TVar*> list;
-            recurseVariablesDown(var, list, 0);
-            QListIterator<TVar*> it2(list);
-            while (it2.hasNext()) {
-                TVar* var2 = it2.next();
-                if (!showHiddenVars && vu->isHidden(var2)) {
+            recursiveSearchVariables(var, list, false);
+            QListIterator<TVar*> itVarDecendent(list);
+            while (itVarDecendent.hasNext()) {
+                TVar* varDecendent = itVarDecendent.next();
+                if (!showHiddenVars && vu->isHidden(varDecendent)) {
                     continue;
                 }
-                QTreeWidgetItem* pItem2;
-                if (!var2->getName().isEmpty() && (var2->getName().indexOf(s, 0, Qt::CaseInsensitive) != -1)) {
+
+                QTreeWidgetItem* pItem;
+                QTreeWidgetItem* parent = nullptr;
+                QString name = varDecendent->getName();
+                QString value = varDecendent->getValue();
+                QStringList idStringList = vu->shortVarName(varDecendent);
+                QString idString;
```

Comment:
`idString` is used as the "Name" part for a variable search result and it should produce something that the user can use themselves to access the "value" of the item as it gives the "full" path to the item. 😁   Testing/confirming by reviewers would be welcome but I think it is _good to go_...